packages/typespec-ts/src/operationUtil.ts

```diff
@@ -74,11 +75,15 @@ export function isDefinedStatusCode(statusCode: StatusCode) {
   return statusCode !== "*";
 }
 
-export function isBinaryPayload(body: Type, contentType: string) {
+export function isBinaryPayload(
```

Comment:
Now that TypeSpec supports [@encode ](https://microsoft.github.io/typespec/standard-library/built-in-decorators#@encode) I wonder if we can leverage it to get rid of this complex logic.

Reply:
https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/src/modelUtils.ts#L837-L847 Currently we already have this implemented.

This check helps on binary body not byte in a normal model.